contracts/src/v0.8/dev/ArbitrumValidator.sol

```diff
@@ -0,0 +1,140 @@
+   */
+  constructor(
+    address aggregatorAddress,
+    address inboxAddress,
+    address flagAddress,
+    address gasConfigAccessController,
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  ) {
+    require(flagAddress != address(0), "Invalid Flags contract address");
+    s_arbitrumInbox = ArbitrumInboxInterface(inboxAddress);
+    s_gasConfigAccessController = SimpleWriteAccessController(gasConfigAccessController);
+    s_flagsAddress = flagAddress;
+    _setGasConfiguration(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+
+    this.addAccess(aggregatorAddress);
+  }
+  
+  fallback() external payable {}
+
+  function withdrawFunds() 
+    external 
+    onlyOwner() 
+  {
+    address payable to = payable(msg.sender);
+    to.transfer(address(this).balance);
+  }
+
+  function withdrawFundsTo(
+    address payable to
+  ) 
+    external
+    onlyOwner() 
+  {
+    to.transfer(address(this).balance);
+  }
+
+  function setGasConfiguration(
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  )
+    external
+  {
+    require(s_gasConfigAccessController.hasAccess(msg.sender, msg.data), "Only gas configuration admin can call");
+    _setGasConfiguration(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+  }
+
+  function validate(
+    uint256 /* previousRoundId */,
+    int256 /* previousAnswer */,
+    uint256 /* currentRoundId */,
+    int256 currentAnswer
+  ) 
+    external 
+    override
+    checkAccess() 
+    returns (bool) 
+  {
+    // Avoids resending to L2 the same tx on every call
+    if (s_lastChangedAnswer == currentAnswer) {
+      return true;
+    }
+    s_lastChangedAnswer = currentAnswer;
+    bytes memory data = currentAnswer == 1 ? abi.encodeWithSelector(RAISE_SELECTOR, s_arbitrumFlag) : abi.encodeWithSelector(LOWER_SELECTOR, s_arbitrumFlag);
+
+    s_arbitrumInbox.createRetryableTicket{value: s_gasConfig.gasCostL2}(
+      s_flagsAddress, 
+      0, // L2 call value
+      s_gasConfig.maximumSubmissionCost, // Max submission cost of sending data length
+      s_gasConfig.refundableAddress, // excessFeeRefundAddress
+      s_gasConfig.refundableAddress, // callValueRefundAddress
+      s_L2GasLimit,
+      s_gasConfig.maximumGasPrice, 
+      data
+    );
+    return true;
+  }
+  
+  event GasConfigurationSet(
+    uint256 maximumSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  );
+  
+  function _setGasConfiguration(
+    uint256 _maximumSubmissionCost,
+    uint32 _maximumGasPrice,
+    uint256 _gasCostL2,
+    address _refundableAddress
+  ) internal {
+    uint256 minGasCostValue = _maximumSubmissionCost + s_L2GasLimit * _maximumGasPrice;
+    require(_gasCostL2 >= minGasCostValue, "The gas cost value provided is too low to cover the L2 transactions");
```

Comment:
Try to limit all revert string to max 32 chars, so they fit into one 32 byte EVM word and we save on gas.

Reply:
This will be removed, as `gasCostL2` will probably need to be `zero` on L1

Reply:
IMHO this check makes sense if `gasCostL2 > 0` (L1 payments), no need to check if 0 (L2 payments).

Reply:
Ok, that makes sense